File: 2793-count-the-number-of-complete-components/2793-count-the-number-of-complete-components.py
class Solution(object):
    def countCompleteComponents(self, n, edges):
        """
        :type n: int
        :type edges: List[List[int]]
        :rtype: int
        """
        adj = {}
        for i in range(n):
            adj[i] = []
        
        for edge in edges:
            u = edge[0]
            v = edge[1]

            adj[u].append(v)
            adj[v].append(u)

        cnt = 0
        vis = [False for _ in range(n)]

        for i in range(n):
            if not vis[i]:
                comp = []
                q = deque()

                vis[i] = True
                q.append(i)
                
                while q:
                    node = q.popleft()
                    comp.append(node)

                    for neigh in adj[node]:
                        if not vis[neigh]:
                            vis[neigh] = True
                            q.append(neigh)
                flag = True
                for ele in comp:
                    if len(adj[ele]) != len(comp) - 1:
                        flag = not flag
                        break
                if flag:
                    cnt += 1
        

        #print(adj)
        return cnt

File: 2793-count-the-number-of-complete-components/2793-count-the-number-of-complete-components.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number}
 */
var countCompleteComponents = function(n, edges) {
    const adj = Array(n).fill().map(() => []);

    for (let [u, v] of edges) {
        adj[u].push(v);
        adj[v].push(u); 
    }

    let cnt = 0;
    const vis = new Array(n).fill(false);

    for (let i = 0; i < n; i++) {
        if (!vis[i]) {
            const comp = [];
            const queue = [i];
            vis[i] = true;

            while (queue.length > 0) {
                const node = queue.shift();
                comp.push(node);

                for (let neigh of adj[node]) {
                    if (!vis[neigh]) {
                        vis[neigh] = true;
                        queue.push(neigh);
                    }
                }
            }

            let flag = true;
            for (let ele of comp) {
                if (adj[ele].length !== comp.length - 1) {
                    flag = false;
                    break;
                }
            }
            if (flag) 
                cnt++;
        }
    }

    return cnt;
};
